webui/react/src/hooks/useModal/Workspace/useModalWorkspaceAddMember.tsx

```diff
@@ -128,21 +133,32 @@ const useModalWorkspaceAddMember = ({ addableUsersAndGroups, onClose }: Props):
         });
       }
     }
-  }, [form, selectedOption]);
+  }, [form, selectedOption, workspaceId]);
 
   const modalContent = useMemo(() => {
     return (
       <div className={css.base}>
-        <Form autoComplete="off" form={form} layout="vertical">
+        <Form
+          autoComplete="off"
+          form={form}
+          layout="vertical"
+          onValuesChange={async () => await form.validateFields()}>
           <Form.Item
             label="User or Group"
             name="userOrGroupId"
             rules={[{ message: 'User or group is required ', required: true }]}>
             <Select
               filterOption={handleFilter}
               options={addableUsersAndGroups.map((option) => ({
-                label: getName(option),
-                value: getIdFromUserOrGroup(option),
+                label: isUser(option) ? (
+                  getName(option)
+                ) : (
+                  <span>
+                    {getName(option)}&nbsp;&nbsp;
+                    <Icon name="group" />
+                  </span>
+                ),
+                value: (isUser(option) ? 'u_' : 'g_') + getIdFromUserOrGroup(option),
```

Comment:
This is fine for now but just in general instead of ad-hoc serialization like this it's better to JSON serialize or even better expand the component to handle many types transparently.